Framework/Algorithms/src/DetectorEfficiencyCorUser.cpp

```diff
@@ -105,24 +103,22 @@ void DetectorEfficiencyCorUser::exec() {
 
 /**
  * Apply the detector efficiency to a single spectrum
- * @param numberOfChannels Number of channels in a spectra (nbins - 1)
- * @param yIn spectrum counts
- * @param eIn spectrum errors
+ * @param nChans Number of channels in a spectra (nbins - 1)
  * @param effVec efficiency values (to be divided by the counts)
- * @param yOut corrected spectrum counts
- * @param eOut corrected spectrum errors
+ * @param histogram uncorrected histogram
+
+ * @returns corrected histogram
  */
-void DetectorEfficiencyCorUser::applyDetEfficiency(
-    const size_t numberOfChannels, const Mantid::HistogramData::HistogramY &yIn,
-    const Mantid::HistogramData::HistogramE &eIn, const MantidVec &effVec,
-    Mantid::HistogramData::HistogramY &yOut,
-    Mantid::HistogramData::HistogramE &eOut) {
-
-  for (unsigned int j = 0; j < numberOfChannels; ++j) {
-    // xOut[j] = xIn[j];
-    yOut[j] = yIn[j] / effVec[j];
-    eOut[j] = eIn[j] / effVec[j];
+Histogram DetectorEfficiencyCorUser::applyDetEfficiency(
+    const size_t nChans, const MantidVec &effVec, const Histogram &histogram) {
+  Histogram outHist(histogram);
+
+  for (unsigned int j = 0; j < nChans; ++j) {
+    outHist.mutableY()[j] = histogram.y()[j] / effVec[j];
+    outHist.mutableE()[j] = histogram.e()[j] / effVec[j];
```

Comment:
`mutable?()` in inner loop, 👎  (please change ;) )

Apart from that, considering performance (maybe not a major issue here, but in the long run it pays off the pay attention to these things), consider what happens here:

1. `mutableY()` triggers a copy.
2. You set the values in the copy to the value in the original divided by some value. The resulting data movement is thus: load original, load copy, load scale, store copy, i.e., 4 * 8 Byte for each value in the Y output. If you instead scale the copy directly you avoid loading the original, i.e., the CPU has to move only 3 * 8 Byte.